pubsub/pubsub.go

```diff
@@ -22,16 +22,30 @@ import (
 
 // PubSub is the interface for message buses.
 type PubSub interface {
+	BulkMessager
 	Init(metadata Metadata) error
 	Features() []Feature
 	Publish(req *PublishRequest) error
 	Subscribe(ctx context.Context, req SubscribeRequest, handler Handler) error
 	Close() error
 }
 
+// BulkMessager is the interface defining BulkPublish and BulkSubscribe definitions for message buses
+type BulkMessager interface {
+	BulkPublish(req *BulkPublishRequest) (BulkPublishResponse, error)
+	BulkSubscribe(ctx context.Context, req SubscribeRequest, handler BulkHandler) (BulkSubscribeResponse, error)
+}
+
 // Handler is the handler used to invoke the app handler.
 type Handler func(ctx context.Context, msg *NewMessage) error
 
+// BulkHandler is the handler used to invoke the app handler.
+// It returns first type as []BulkSubscribeResponse which represents status per message - if not nil,
+// broker can take appropriate action accordingly.
+// Second return type is error which if not nil, reflects that there was an issue with
+// the whole bulk event and nothing could be sent ahead.
+type BulkHandler func(ctx context.Context, msg *BulkMessage) ([]BulkSubscribeResponse, error)
```

Comment:
The return type here is wrong - it should be an array of BulkSubscriberesponseEntry or simply BulkSubscribeResponse. Will change it - just testing few things out.